pennylane/estimator/resource_tracking.py

```diff
@@ -0,0 +1,350 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+r"""Core resource tracking logic."""
+from collections import defaultdict
+from collections.abc import Callable, Iterable
+from functools import singledispatch, wraps
+
+from pennylane.operation import Operation
+from pennylane.queuing import AnnotatedQueue, QueuingManager
+from pennylane.wires import Wires
+
+from .resource_config import ResourceConfig
+from .resource_operator import (
+    CompressedResourceOp,
+    GateCount,
+    ResourceOperator,
+)
+from .resources_base import DefaultGateSet, Resources
+from .wires_manager import Allocate, Deallocate, WireResourceManager
+
+# pylint: disable=protected-access,too-many-arguments
+
+
+def map_to_resource_op():  # TODO: Import this function instead when the mapping PR is merged
+    """Maps an instance of :class:`~.Operation` to its associated :class:`~.pennylane.labs.resource_estimation.ResourceOperator`."""
+
+
+def estimate(
+    obj: ResourceOperator | Callable | Resources | list,
+    gate_set: set | None = None,
```

Comment:
Can you update the type hints and documentation accordingly?
```suggestion
    gate_set: set[str] | None = None,
```